libcalico-go/lib/epstatusfile/status_file_watcher_test.go

```diff
@@ -125,6 +153,9 @@ var _ = Describe("Workload endpoint status file watcher test", func() {
 
 		filePath := filepath.Join(statusDir, "pod1")
 		Eventually(haveEvents, "5s", "1s").WithArguments(filePath, []string{"create"}).Should(BeTrue())
+		Eventually(r.inSyncEvents).ShouldNot(BeEmpty())
```

Comment:
Eventually with a slice is a smell.  Due to Go's value semantics, this is equivalent to 
```
s := r.inSyncEvents
Eventually(s).ShouldNot(...)
```
So, apart from racing with writes to `inSyncEvents` the code doesn't actually re-read the slice each time through Eventually's check loop.

You should:
- Protect `inSyncEvents` with a mutex
- Pass a func to Eventually

Reply:
Locking everything in `eventRecorder` now